src/Forms/NumericField.php

```diff
@@ -232,7 +244,16 @@ public function getSchemaValidation()
      */
     public function dataValue()
     {
-        return $this->cast($this->value);
+        $value = $this->getValue();
+        if ($value === '') {
+            return null;
+        } elseif ($value === false) {
+            return 0;
+        }
```

Comment:
1. When would the value be explicitly `false`?
1. Why do we return `0` here, but cast returns `false`?

Reply:
false $value happens when the localisation formatter fails to parse the in setSubmittedValue()

dataValue() is passed to the DBField, we do not want to be settings the DBField value to false. 0 is appropriate.

Reply:
We shouldn't be passing the _wrong_ value into `DBInt`. If someone sets an invalid value to the form field, we shouldn't be replacing whatever value used to be stored in the database with `0`

 Let it return false from here, and the validation logic in `DBInt` will pick it up.

Reply:
That's true. Won't necessarily be DBInt as it could be any DBField e.g. DBVarchar. However NumericField::validate() will prevent code from even reaching this point so it doesn't really matter. I've expanded on the inline comment to explain this. In hindsight the only reason I was returning zero here was to get an existing unit test to pass, I've now updated the unit test to expect false instead of 0

Reply:
> However NumericField::validate() will prevent code from even reaching this point so it doesn't really matter.

It won't in a scenario where someone is calling methods directly - and in that case setting `0` instead of letting `DBField` validation kick in would have been inappropriate.